src/app/content/utils/seoUtils.ts

```diff
@@ -0,0 +1,22 @@
+import { assertDefined } from '../../utils';
+import { Book, Page } from '../types';
+import {
+  archiveTreeSectionIsChapter,
+  findArchiveTreeNode,
+  getArchiveTreeNodeTitle,
+  getArchiveTreeSectionNumber,
+} from './archiveTreeUtils';
+
+export const createTitle = (page: Page, book: Book): string => {
+  let node = assertDefined(findArchiveTreeNode(book.tree, page.id), `couldn't find node for a page id: ${page.id}`);
+  const nodeTitle = getArchiveTreeNodeTitle(node);
+  const title = `${nodeTitle} - ${book.title} | OpenStax`;
+  while (node && node.parent) {
```

Comment:
according to the requirements, we should include the chapter prefix only if the page doesn't have a section number of its own

Reply:
This wasn't clear, because AC says:
> - Pages within a chapter container have this template for the tag: Ch. [#][Page Title] - [Book] | OpenStax
> - Pages NOT within a chapter container should have this template for the tag: [Page Title] - [Book] | OpenStax 

and sections with numbers may also be in the chapter.

Reply:
the spreadsheet is more clear, i'll have kim update the AC